internal/server/users.go

```diff
@@ -83,24 +83,22 @@ func GetUser(c *gin.Context, r *api.GetUserRequest) (*api.User, error) {
 // CreateUser creates a user with the Infra provider
 func (a *API) CreateUser(c *gin.Context, r *api.CreateUserRequest) (*api.CreateUserResponse, error) {
 	rCtx := getRequestContext(c)
-	user := &models.Identity{Name: r.Name}
 
-	// infra identity creation should be attempted even if an identity is already known
-	identities, err := access.ListIdentities(rCtx, data.ListIdentityOptions{ByName: r.Name})
-	if err != nil {
-		return nil, fmt.Errorf("list identities: %w", err)
-	}
-
-	switch len(identities) {
-	case 0:
+	user, err := access.GetIdentity(rCtx, data.GetIdentityOptions{ByName: r.Name, LoadProviders: true})
+	switch {
+	case errors.Is(err, internal.ErrNotFound):
+		user = &models.Identity{Name: r.Name}
 		if err := access.CreateIdentity(rCtx, user); err != nil {
 			return nil, fmt.Errorf("create identity: %w", err)
 		}
-	case 1:
-		user.ID = identities[0].ID
-	default:
-		logging.Errorf("Multiple identities match name %q. DB is missing unique index on user names", r.Name)
-		return nil, fmt.Errorf("multiple identities match specified name") // should not happen
+	case err != nil:
+		return nil, fmt.Errorf("get identities: %w", err)
+	}
+
+	for _, provider := range user.Providers {
```

Comment:
seems like this check for if the user exists in the infra provider would make the most sense as the `default` case in the switch statement above where it doesn't need to run when the user didnt exist before.